contracts/YourContract.sol

```diff
@@ -19,16 +19,29 @@ contract YourContract {
         umbChain = IChain(_umbChainAddress);
     }
 
-    function getLatestNumericFCD(bytes32 _key) public view returns (uint256) {
-        uint256 blockHeight = umbChain.getBlockHeight();
-        console.log("Fetching data from UMB Network for current block height", blockHeight);
+    function getNumericFCD(uint256 _blockHeight, bytes32 _key) public view returns (uint256) {
+        console.log("Fetching data from UMB Network for block height", _blockHeight);
 
-        uint256 value = umbChain.getSingleNumericData(blockHeight, _key);
+        uint256 value = umbChain.getSingleNumericData(_blockHeight, _key);
 
         console.log("key:");
         console.logBytes32(_key);
         console.log("value=", value);
 
         return value;
     }
+
+    function getLatestNumericFCD(bytes32 _key) public view returns (uint256) {
+        uint256 blockHeight = umbChain.getBlockHeight();
+        console.log("current block height:", blockHeight);
+
+        if (umbChain.getBlockVotersCount(blockHeight) > 0) {
+            return getNumericFCD(blockHeight, _key);
+        } else {
+            console.log("there is no consensus yet on this block:", blockHeight);
+            console.log("fetching previous");
```

Comment:
what if there's no consensus in the previous block?
I think we can have a variable in Chain with the latest successful block.

Reply:
there is **always** consensus in previous block (except of block 0 of course)

Reply:
even for such a simple use-case the code looks unclear. Can we encapsulate this if statement inside Chain contract? 
So, we call something like getLastNumericFCD